.travis.yml

```diff
@@ -48,15 +48,18 @@ jobs:
 
       php: 7.1
 
+      env: WITH_LOWEST=true
+
       before_install:
         - source .travis/xdebug.sh
         - xdebug-disable
         - composer validate
 
       install:
         - if [[ "$TRAVIS_PHP_VERSION" == "7.3" ]]; then composer remove --dev friendsofphp/php-cs-fixer; fi
-        - if [ -n "${SYMFONY_VERSION}" ]; then composer require symfony/symfony:${SYMFONY_VERSION} --no-update; fi;
-        - composer update ${COMPOSER_FLAGS} --prefer-dist
+        - if [ -n "${SYMFONY_VERSION}" ]; then composer require symfony/symfony:${SYMFONY_VERSION}; fi;
+        - if [[ "$WITH_LOWEST" == "true" ]]; then composer update --prefer-lowest; fi
```

Comment:
Perhaps we should also check in `composer.lock` and run builds against locked versions of dependencies?

```diff
diff --git a/.gitignore b/.gitignore
index 55cecff..8fd2c28 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,3 @@
-composer.lock
 vendor
 docs/build
 .phpbench.sqlite
diff --git a/.travis.yml b/.travis.yml
index bdbf671..68ba9d3 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -59,6 +59,7 @@ jobs:
         - if [[ "$TRAVIS_PHP_VERSION" == "7.3" ]]; then composer remove --dev friendsofphp/php-cs-fixer; fi
         - if [ -n "${SYMFONY_VERSION}" ]; then composer require symfony/symfony:${SYMFONY_VERSION}; fi;
         - if [[ "$WITH_LOWEST" == "true" ]]; then composer update --prefer-lowest; fi
+        - if [[ "$WITH_LOCKED" == "true" ]]; then composer install; fi
         - if [[ "$WITH_HIGHEST" == "true" ]]; then composer update; fi

       before_script:
@@ -69,6 +70,12 @@ jobs:
         - php bin/phpbench run --report=env --progress=dots --store
         - php bin/phpbench run --iterations=1 --revs=1 --config=extensions/dbal/benchmarks/phpbench.json --progress=dots

+    - <<: *TEST
+
+      php: 7.1
+
+      env: WITH_LOCKED=true
+
     - <<: *TEST

       php: 7.1
@@ -87,6 +94,12 @@ jobs:

       env: WITH_LOWEST=true

+    - <<: *TEST
+
+      php: 7.2
+
+      env: WITH_LOCKED=true
+
     - <<: *TEST

       php: 7.2
@@ -99,6 +112,12 @@ jobs:

       env: WITH_LOWEST=true

+    - <<: *TEST
+
+      php: 7.3
+
+      env: WITH_LOCKED=true
+
     - <<: *TEST

       php: 7.3
```

Reply:
hmm, not sure what you mean, we don't commit the `composer.lock`?

Reply:
I would think we should also try and balance the build time against the version coverage. Previously we just had one PHP version to test the lowest dependencies (IIRC) for this reason. So would be happy to keep the matrix to a minimum and just f.e. test the lowest deps against the lowest PHP version.

Is there any difference between `composer install` and `composer update` as far as the build is concerned?

Reply:
There will only be a difference between

```
$ composer install
```

and 

``` 
$ composer update
```

when `composer.lock` is not checked in, as far as I know.

Reply:
oh sorry, I missed the title of the PR :man_facepalming: 

For libraries (or `require` included tools), I never check-in `composer.lock`. What is the reason for doing so here?

Reply:
Hmm, I think it's a good idea to ensure that the build passes at least against the set of dependencies that were used in development, but I guess we don't have to.

Reply:
but that set of dependencies is arbitrary - there are innumerable combinations of dependencies that will be used in the wild. I don't _think_ there is any particular reason to assume that the state of the application used in development is special in any particular way. The Phar is also built  based on the lastest dependencies.

So, would currently prefer to keep the `composer.lock` unversioned :)

Reply:
Done!